File: Experience/Tutorials/Languages/Python/test_number_pos_or_neg.py
while True:
    num_input = input('Input a number if it is positive or negative.(Type x to exit the program)')
    
    if num_input == 'x':
        print('exiting the program...')
        break
    # use this keyword if errors been found
    try:
        num = int(num_input)
            
        if num > 0:
            print('The number is positive.')
        elif num < 0:
            print('The number is negative.')
        elif num == 0:
            print('The number is zero.')
    except ValueError:
        print('Invalid input. Please enter a number.')
        

File: Experience/Tutorials/Languages/Java/Tutorials/Translation/TestNumberPosOrNeg.java
package Java.Tutorials.Translation;

import java.util.Scanner;

public class TestNumberPosOrNeg {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        int num;
        boolean continueCalc = true;

        while (continueCalc) {
            try { 
        
                System.out.println("Input a number if it is positive or negative.(Type x to exit)");
        
                num = scanner.nextInt();
                
                if (num > 0) {
                    System.out.println("The number is positive");
                }
        
                else if (num < 0) {
                    System.out.println("The number is negative");
                }
        
                else if (num == 0) {
                    System.out.println("The number is zero");
                }
                else if (num == (char) 'x') {
                    System.out.println("Exiting the program...");
                    continueCalc = false;
                    break;
                }
                else {
                    System.out.println("Invalid input. ");
                }
        
                    
            }    
             catch (NumberFormatException e) {
                 System.out.println("Invalid input. Please enter a number.");
                }
        }

        

        
    }
}

/*
 *    
 */